### PR TITLE
RESTWS-877: Concept attribute type retire test | Endpoint

### PR DESCRIPTION
**Pull request by Muta-Jonathan**
## Description of what I changed
Added retire Test for Concept attribute type ~ Endpoint

## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-877

### CODE REVIEW

**Comment by Muta-Jonathan** [comment]
cc @dkayiwa @kdaud @sherrif10

**Review by kdaud — APPROVED** [review]
cc: @dkayiwa

**Comment by dkayiwa** [comment]
@Muta-Jonathan can you also assert the date retired and retired by fields?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java [hunk 1]
```diff
@@ -121,4 +121,21 @@ public void shouldGetAConceptAttributeTypeByUuid() throws Exception {
     	Assert.assertEquals(conceptAttributeType.getUuid(), PropertyUtils.getProperty(result, "uuid"));
     	Assert.assertEquals(conceptAttributeType.getName(), PropertyUtils.getProperty(result, "name"));
     }
+
+    @Test
+    public void shouldRetireAConceptAttributeType() throws Exception {
+        final String UUID = "9516cc50-6f9f-11e0-8414-001e378eb67e";
+        ConceptAttributeType conceptAttributeType = service.getConceptAttributeTypeByUuid(UUID);
+        Assert.assertNotNull(conceptAttributeType);
+        Assert.assertFalse(conceptAttributeType.isRetired());
```
Comment (Irenyak1) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java:130]: Thanks @Muta-Jonathan . This is great!. Have you tried looking into using getRetired() as opposed to using isRetired()?
The later method seems to be deprecated. Could this [document](https://docs.openmrs.org/doc/org/openmrs/BaseOpenmrsMetadata.html) be helpful?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java [hunk 2]
```diff
@@ -121,4 +121,21 @@ public void shouldGetAConceptAttributeTypeByUuid() throws Exception {
...
+
+        MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + UUID);
+        req.addParameter("!purge", "");
+        req.addParameter("reason", "let it retire for a time");
+        handle(req);
+
+        conceptAttributeType = service.getConceptAttributeTypeByUuid(UUID);
+        Assert.assertTrue(conceptAttributeType.isRetired());
```
Comment (Irenyak1) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java:138]: @Muta-Jonathan  you could do the same here.

**Review by Irenyak1 — COMMENTED** [review]
@Muta-Jonathan  catch the comments :-)

**Comment by Muta-Jonathan** [comment]
cc @dkayiwa

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java (hunk 2 above)
Comment (Muta-Jonathan) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java]: Thanks @Irenyak1...have updated it

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java (hunk 1 above)
Comment (dkayiwa) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java]: Thank you so much @Irenyak1 for the wonderful review. Please keep it up! 👍

**Comment by Muta-Jonathan** [comment]
Thanks @dkayiwa

**Pull request merged**